Bowling.java

```diff
@@ -154,21 +154,44 @@ private Boolean isTurkey() {
 	private Boolean isSingleStrike() {
 		return secondToLastFrame.isPresent() && 
 		       secondToLastFrame.get().isStrike() && 
-		       !currentFrame.isStrike();
+		       !currentFrame.isStrike() &&
+					 (!thirdToLastFrame.isPresent() ||
+					   thirdToLastFrame.isPresent() && !thirdToLastFrame.get().isStrike());
 	}
```

Comment:
Your suggestion makes sense, but I confess to feeling like it's abstracting on the wrong patterns.  It is literally correct that we need to consider those number of strikes, but what I like about the other flow that I added it the reliance on `getFrameScore()`.  Also, I could envision a world where we ask the frame if it cares about any more bowls - and let it understand the significance of spare and strike...

In the end I don't think we're hurting our selves either way and since I only have some half-baked ideas of where to go I think it we should follow your idea.

I should also note that I'm embarrassed to have kept the `System.out.println("scoring...");` line in there :)